pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -220,3 +224,53 @@ func draAdminAccessFeatureInUse(claim *resource.ResourceClaim) bool {
 
 	return false
 }
+
+func dropDisabledDRAResourceClaimDeviceStatusFields(newClaim, oldClaim *resource.ResourceClaim) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DRAResourceClaimDeviceStatus) {
```

Comment:
```suggestion
        // drop resourceClaim.Status.Devices field if feature gate is not enabled and it was not in use
        isDRAResourceClaimDeviceStatusInUse := (oldClaim != nil && oldClaim.Status.Devices != nil)
	if !utilfeature.DefaultFeatureGate.Enabled(features.DRAResourceClaimDeviceStatus)  &&  !isDRAResourceClaimDeviceStatusInUse {
```

Reply:
As with the other place, I'd use a `len()` check rather than `== nil`.